### PR TITLE
Refactor containers to use hooks

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Related to https://github.com/tektoncd/dashboard/issues/1842
### **Note:** ignore whitespace for an easier review

Refactor the App, About, Conditions, and Condition containers to
be functional components using React hooks instead of being class
components.

This change is a prerequisite for adopting react-query and other
libraries which provide a hooks-based API.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/App/App.js [hunk 1]
```diff
@@ -197,281 +161,299 @@ export /* istanbul ignore next */ class App extends Component {
...
+    } catch (error) {
+      console.error(error); // eslint-disable-line no-console
+      setLoadingConfigError(error);
+      setIsLoadingConfig(false);
+      setShowLoadingState(false);
+    }
   }
+
+  useEffect(() => {
+    if (webSocketConnected !== false) {
```
Comment (steveodonovan) [inline src/containers/App/App.js:183]:
```
if (webSocketConnected) {
...
```

File: src/containers/Condition/Condition.js [hunk 2]
```diff
@@ -21,116 +21,95 @@ import {
...
-    } = prevMatch.params;
-
-    if (
-      namespace !== prevNamespace ||
-      conditionName !== prevConditionName ||
-      (webSocketConnected && prevWebSocketConnected === false)
-    ) {
-      this.fetchData();
-    }
+function AdditionalContent({ condition, intl }) {
```
Comment (steveodonovan) [inline src/containers/Condition/Condition.js:29]: Naming nit, additional content isn't obvious, `ConditionParameters` or similar maybe ?

File: src/containers/About/About.js [hunk 3]
```diff
@@ -84,143 +93,139 @@ export /* istanbul ignore next */ class About extends Component {
...
+      header: intl.formatMessage({
+        id: 'dashboard.tableHeader.value',
+        defaultMessage: 'Value'
+      })
+    }
+  ];
+
+  const getRow = (property, value) => {
+    const displayValue = getDisplayValue(value);
+    return displayValue
```
Comment (steveodonovan) [inline src/containers/About/About.js:115]: `displayValue && `{ id: property, property, value: displayValue }`

File: src/containers/App/App.js (hunk 1 above)
Comment (AlanGreene) [reply src/containers/App/App.js:183]: By default `useEffect` is called after every render (including the first one after the component is mounted). We're limiting that here by passing `webSocketConnected` in runs after the first render and then only after subsequent renders where the value of `webSocketConnected` has changed.

[`webSocketConnected` is initially `null`](https://github.com/tektoncd/dashboard/blob/4d23fc20367beb5e4370ca89f28dbef0486d0d12/src/reducers/notifications.js#L15) until we attempt to connect. We want to call `fetchConfig` when the `App` component mounts, regardless of the websocket connection status, and later only on reconnect.

As part of the switch to `react-query` we'll also be ditching these status booleans and manual checks, letting react-query handle most of the reconnect / retry logic for us.

File: src/containers/Condition/Condition.js (hunk 2 above)
Comment (AlanGreene) [reply src/containers/Condition/Condition.js]: updated

File: src/containers/About/About.js (hunk 3 above)
Comment (AlanGreene) [reply src/containers/About/About.js]: updated

**Review by steveodonovan — APPROVED** [review]
/approve
/lgtm

**Comment by AlanGreene** [comment]
First part of https://github.com/tektoncd/dashboard/issues/2000

**Pull request merged**